### DISCUSSION TITLE
Scalability expectations

### DISCUSSION THREAD

**Discussion by jagjordi** [immich-app/immich#1180] [General]
How much is immich able to handle in theory? Order of magnitude approximation, thousands, tens of thousands, hundreds of thousands, milions? Of pictures. Also what is the limiting factor? Is it disk speed? Network bandwidth? CPU power?

**Comment by alextran1502** [comment]
The way we build Immich is to prioritize lazy-loading so if you are not scrolling to that specific time block, location on the timeline, nothing else should be fetched. So in theory it should be in the magnitude of millions before we reach some disk seek operation bottleneck, however we have not had any reports or any data of when that would happen yet.
